aee.js

```diff
@@ -72,3 +72,14 @@ Array.prototype.count = function (spec) {
   }
   return result
 }
+
+Array.prototype.index = function (spec) {
+  const len = this.length
+  if (len === 0 || spec === undefined) return -1
+  let searchSpec = (typeof spec === 'function') ? spec
```

Comment:
I've seen this too many times, if I were on your shoes probably at this point I would have something like.

```js
function nfn(spec, fn) {
    return (typeof spec === 'function') ? spec : fn;
}

function foo(spec) {
    let searchSpec = nfn(spec, value => JSON.stringify(value) === JSON.stringify(spec))
}
```

Just saying...